sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/component.py

```diff
@@ -510,6 +510,16 @@ def _get_telemetry_values(self, *args, **kwargs):  # pylint: disable=unused-argu
 
     def __call__(self, *args, **kwargs) -> [..., Union["Command", "Parallel"]]:
         """Call ComponentVersion as a function and get a Component object."""
+        if args:
+            # raise clear error message for unsupported positional args
+            msg = f"Component function doesn't support positional arguments, got {args} for {self.name}. " \
+                  f"Please use keyword arguments instead."
```

Comment:
I think the reason we need to refine this is user might have difficulty to understand what's 
- positional arguments
- keyword arguments 

Please use keyword arguments instead, e.g. user_func(input1=xx)

Reply:
Added example for component function call, but we can not get user's component func name so I used fixed "component_func"